contracts/Core/StakeManager.sol

```diff
@@ -103,21 +102,22 @@ contract StakeManager is Initializable, ACL, StakeStorage, StateManager, Pause {
         uint256 amount
     ) external initialized checkEpochAndState(State.Commit, epoch, parameters.epochLength()) whenNotPaused {
         require(stakers[stakerId].acceptDelegation, "Delegetion not accpected");
-        // Step 1:  Razor Token Transfer : Amount
-        require(razor.transferFrom(msg.sender, address(this), amount), "RZR token transfer failed");
+        require(stakers[stakerId].tokenAddress != address(0x0), "Staker has not staked yet");
+        emit Delegated(msg.sender, epoch, stakerId, stakers[stakerId].stake, block.timestamp);
```

Comment:
Hey by moving this up,
It will emit old data, as the stake is being changed later
Please add it after that
And please check in other places as well if by moving anything up or down, its calculating on old data

Reply:
Ideally, Delegated should only emit the amount that has been delegated instead of whole stake. we can replace that with amount.

Reply:
Hmm nope, the whole stake is only being used in RazorScan

Reply:
I think we should rename to updatedStake if we want whole stake. Delegated will misguide user

Reply:
hmm not needed 
as in event delegated, we are maintaining it as newStake

```
 event Delegated(address delegator, uint32 epoch, uint32 indexed stakerId, uint256 newStake, uint256 timestamp);
 ```

Reply:
fixed